### ISSUE TITLE
Describe the process of adding new skins

### ISSUE THREAD

**Issue by sskorol** [robotics-upo/hunav_gazebo_wrapper#4]
Hi @noeperez @roberottt,

I'm trying to add a new model to a simulation. Here are the steps I've completed so far:
- built it via `MakeHuman` app
- animated in `Mixamo`
- exported to `.dae` and `.bvh` via `Blender` and put them to `models` folder
- added new behaviors to `Agent.msg`
- extended `skin_filename` and `animation_filename` with newly added items
- added `hunav_msgs` handlers for the newly added behaviors

When I start all the services via launch file, I see the following error:

![Screenshot from 2022-11-16 04-39-50](https://user-images.githubusercontent.com/6638780/202070356-b2acb565-34ea-48b1-9fa3-5ea2850bcfb2.png)

As it comes from Gazebo, I'm not really sure how exactly I should fix it. Where are these active/inactive animations are set?

Would be greatly appreciated any thoughts.

**Comment by noeperez** [comment]
Hi!
To create new agent models is not trivial. We struggled a bit to achieve it, but we did it.
We will update the wrapper with a set of new models soon (man, women, children, ..). They have a better look than the default bald agents.
We will also upload a guide of the steps that we followed to create them. (We are not expert in modelling, so probably an expert in this field could reach a better and easier way).

**Comment by noeperez** [comment]
Sure! 
We use the "active" state to define the animation that the agent should perform while reacting to the presence of the robot. 
The "no_active" is the animation while walking without the presence of the robot.

**Comment by sskorol** [comment]
To be more specific, I just added new .dae and .bvh files:

![Screenshot from 2022-11-16 12-10-34](https://user-images.githubusercontent.com/6638780/202152338-8fe84301-b8e9-47ae-83a1-e36ac4470392.png)

Plus corresponding processing logic:
![Screenshot from 2022-11-16 12-11-55](https://user-images.githubusercontent.com/6638780/202152663-f4e4ecb4-b31a-470a-b734-a67c90cff061.png)
And new behaviors:
![Screenshot from 2022-11-16 12-12-45](https://user-images.githubusercontent.com/6638780/202152901-eef6cda5-4a7b-4147-a9d0-e6cc70005b49.png)

**Comment by noeperez** [comment]
To replace an animation is not complex, while the animation is compatible with the skeleton of the model (related to your first problem). 
If you also want to add a new behavior, that is not trivial, and more work is required since the behaviors are controlled by the behaviors trees. You need to program a new behavior tree and register it.

**Comment by sskorol** [comment]
But I generated both .dae and .bvh from a single .fbx. So animation should 100% match the .dae, as it works in Blender and the other tools. So I guess there's something more on the code level where some default animation is used with my model (which is incompatible with the existing animations).

**Comment by sskorol** [comment]
No, I'm stuck. Seems like there should be some special model/animation configuration to play well with Gazebo. There are a bunch of ways how you can export data from Blender. And all the tutorials are pretty much outdated. I don't know why it doesn't work. Looking forward to receiving some instructions from you. Thanks.

**Comment by noeperez** [comment]
Building a functional model with animation to play in Gazebo properly is quite tricky. 
Again, as I commented in the first response, we will upload new models and to try to provide a guide of the steps we followed.

**Comment by sskorol** [comment]
Thanks, do you have any raw estimate? So that I'd understand if I want to spend more time on investigation or just wait for your update. You know, if it's a week, for example, I can wait for sure. If a month, I'd likely find a solution myself.

**Comment by noeperez** [comment]
Probably, we could upload something next week.

**Comment by noeperez** [comment]
We have just added 5 new human models. 
We will provide the step we followed to create them next week.

**Comment by sskorol** [comment]
@noeperez top! Will try later today. Thanks!

**Comment by sskorol** [comment]
New models work for me. I'm looking forward to getting the instructions.

**Comment by roberottt** [comment]
Hi!
We have just added the instructions for custom models.

**Comment by sskorol** [comment]
@roberottt excellent, thanks!

I'm wondering if you also tried to change the animation. I made many attempts, but it seems like Blender exports a broken .bvh file, which can't be correctly handled by Gazebo. Or maybe I just missed something.

**Comment by noeperez** [comment]
Hi!

The models built like that are compatible with all the .bvh animations from [here](https://sites.google.com/a/cgspeed.com/cgspeed/motion-capture/the-motionbuilder-friendly-bvh-conversion-release-of-cmus-motion-capture-database). We just associated the proper animations, for the "active" and "non-active" modes, to the behavior indicated.

**Comment by sskorol** [comment]
@noeperez yes, I know that dataset. However, I'm asking if you tried some custom animations built with e.g. [Mixamo](https://www.mixamo.com/)? It's a very simple and powerful UI tool. You can import your .fbx from MakeHuman, add any animation, then export back to .fbx and import it to Blender with a fully working animation. What I'm stuck at within this process is splitting .fbx onto .dae and .bvh. If I use only .dae, it seems to work well in Gazebo w/o animation. But when I try to export .bvh, it breaks the model: there's a monster from a horror movie walking in Gazebo. :) 
So wondering if you'd have more luck than me with this process. As it should be a lot simpler. I likely missed the valid export options for the target .bvh.

**Comment by sskorol** [comment]
I tried to follow the instructions but got a broken model at the end. Have you tried adding such parts as ponytails, eyelashes, and eyebrows or changing the size of the body? It seems like `walk.dae` have a bit different skeleton compared to what MakeHuman produces. I list I see some significant diffs in some bones' positions. I don't know if that's the reason, but if I just scale a skeleton and leave everything else as is, some parts of the body might not be aligned or be even split.

**Comment by noeperez** [comment]
We have the same problems. When merging the skeleton of the walk.dae with the skin of a new model not always fits perfectly, and some bones positions and connections must be adjusted manually. 
As we say in the document, the 3D modelling is not our field, so that's the best we have achieved for the moment.

**Comment by sskorol** [comment]
Anyway, thanks for your effort!
